limitador/src/storage/mod.rs

```diff
@@ -121,10 +121,12 @@ impl Storage {
 
     pub fn check_and_update(
         &self,
-        counters: HashSet<Counter>,
+        counters: &mut Vec<Counter>,
```

Comment:
Is there a reason you made this a `Vec`? Cause it eventually gets sorted?

Reply:
and HashSet does not have a iter_mut() method.

Reply:
Ah, yeah of course it doesn't... But given the performance improvement avoiding to dedup as we put them in a Set, I'd say this is fine, users "just" need to make sure they don't provide dups...

Reply:
Should we dedup in the counters_that_apply function?

Reply:
No, I think the perf gain is worth it, it'd be weird someone feeds in dups anyways...